srtcore/packetfilter_api.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * SRT - Secure, Reliable, Transport
+ * Copyright (c) 2018 Haivision Systems Inc.
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * 
+ */
+
+#ifndef INC__PACKETFILTER_API_H
+#define INC__PACKETFILTER_API_H
+
+enum SrtPktHeaderFields
+{
+    SRT_PH_SEQNO = 0, //< sequence number
+    SRT_PH_MSGNO = 1, //< message number
+    SRT_PH_TIMESTAMP = 2, //< time stamp
+    SRT_PH_ID = 3,  //< socket ID
+
+    // Must be the last value - this is size of all, not a field id
+    SRT_PH__SIZE
+};
+
+
+enum SRT_ARQLevel
+{
+    SRT_ARQ_NEVER,   //< Never send LOSSREPORT
+    SRT_ARQ_ONREQ,  //< Only record the loss, but report only those that are returned in receive()
+    SRT_ARQ_ALWAYS, //< always send LOSSREPORT immediately after detecting a loss
+};
+
+
+struct SrtFilterConfig
+{
+    std::string type;
+    std::map<std::string, std::string> parameters;
+};
+
+struct SrtFilterInitializer
+{
+    SRTSOCKET socket_id;
+    int32_t snd_isn;
+    int32_t rcv_isn;
+    size_t payload_size;
+};
+
+struct SrtPacket
+{
+    uint32_t hdr[SRT_PH__SIZE];
+    char buffer[SRT_LIVE_MAX_PLSIZE];
```

Comment:
`SRT_LIVE_MAX_PLSIZE` why limit to live payload size? What if someone wants to transmit RTP over SRT?

Reply:
This is the maximum size of a single SRT packet, 1456. The "default live packet size" 1316 is `SRT_LIVE_DEF_PLSIZE`.

Reply:
Rename `SRT_LIVE_MAX_PLSIZE` to `SRT_MAX_PAYLOAD_SIZE` to avoid this confusion in future.